server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -100,16 +104,37 @@ public DatasourcesResource(
   public Response getQueryableDataSources(
       @QueryParam("full") String full,
       @QueryParam("simple") String simple,
+      @QueryParam("nameRegex") String regex,
       @Context final HttpServletRequest req
   )
   {
     Response.ResponseBuilder builder = Response.ok();
-    final Set<DruidDataSource> datasources = authConfig.isEnabled() ?
+    Set<DruidDataSource> datasources = authConfig.isEnabled() ?
                                              InventoryViewUtils.getSecuredDataSources(
                                                  serverInventoryView,
                                                  (AuthorizationInfo) req.getAttribute(AuthConfig.DRUID_AUTH_TOKEN)
                                              ) :
                                              InventoryViewUtils.getDataSources(serverInventoryView);
+    if (!Strings.isNullOrEmpty(regex)) {
+      Set<DruidDataSource> filtered = Sets.newLinkedHashSet();
+      for (String part : regex.split(",")) {
```

Comment:
similar comment here, regarding iterating through the datasources in the outer level